autowarefoundation/autoware.universe
# fix(gnss_poser): no conversion when projection type is local

## Description

Fix an issue where the node dies when the projection type is local.

## Related links

https://github.com/orgs/autowarefoundation/discussions/4157

## Tests performed

1. Add `map_projector_info.yaml` to the your map directory.
    ```
    projector_type: local
    ````
2. Run the following command:
    - Terminal 1: `ros2 launch gnss_poser gnss_poser.launch.xml`
    - Terminal 2: `ros2 topic pub /fix sensor_msgs/msg/NavSatFix`
    - Terminal 3: `ros2 topic pub /map/map_projector_info tier4_map_msgs/msg/MapProjectorInfo "{projector_type: local}"`
3. Check that the node is not died on terminal 1

    - main branch
      ```.
      [gnss_poser-1] [WARN] [1707406335.757544980] [gnss_poser]: map_projector_info has not been received yet. Check if the map_projection_loader is successfully launched.
      [gnss_poser-1] [WARN] [1707406336.757630107] [gnss_poser]: map_projector_info has not been received yet. Check if the map_projection_loader is successfully launched.
      [gnss_poser-1] terminate called after throwing an instance of 'std::invalid_argument'
      [gnss_poser-1]   what():  Invalid map projector type: local. Currently supported types: MGRS, LocalCartesianUTM, and TransverseMercator
      ```
    - PR branch
      ```
      [gnss_poser-1] [WARN] [1707406223.428762538] [gnss_poser]: map_projector_info has not been received yet. Check if the map_projection_loader is successfully launched.
      [gnss_poser-1] [WARN] [1707406224.429247652] [gnss_poser]: map_projector_info has not been received yet. Check if the map_projection_loader is successfully launched.
      [gnss_poser-1] [ERROR] [1707406226.429217000] [gnss_poser]: map_projector_info is local projector type. Unable to convert GNSS pose.
      [gnss_poser-1] [ERROR] [1707406231.429275335] [gnss_poser]: map_projector_info is local projector type. Unable to convert GNSS pose.
      ```

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
